File: Assets/Scripts/ObjectSpawn.js
#pragma strict

public var spawning : boolean = false;
private var newEnemy : GameObject;
private var canSpawn : int = 1;
var player : int = 1;
var minSpawnPerSecond: float = 0.1;
var maxSpawnPerSecond: float = 0.2;
public var Enemy : GameObject[];
var playerStats : PlayerStats;


function Start() {
	playerStats = gameObject.GetComponent("PlayerStats");
}


function FixedUpdate(){
	canSpawn -= 1;
	if (spawning && canSpawn <= 0){
		var randObstacle : int = Random.Range(0,Enemy.length);
		var randPos : int = Random.Range(1,4);
		switch (randPos){
			case 1:
				randPos = -2.5;
				break;
			case 2:
				randPos = 0;
				break;
			case 3:
				randPos = 2.5;
				break;
		}
		newEnemy = Instantiate(Enemy[randObstacle], Vector3(0, randPos, 0), Quaternion.identity);
		if (player == 1){
			var speedMulti = playerStats.P1Speed / 10;
			var oc: ObjectControl = newEnemy.GetComponent("ObjectControl");
			oc.dirRight = false;
		}else {
			speedMulti = playerStats.P2Speed / 10;
			oc = newEnemy.GetComponent("ObjectControl");
			oc.dirRight = true;
		}
		oc = newEnemy.GetComponent("ObjectControl");
		oc.speed *= speedMulti;
		canSpawn = Random.Range((25 / (maxSpawnPerSecond*speedMulti)), (25 / (minSpawnPerSecond*speedMulti)));
	}
}

File: Assets/Scripts/_ObjectSpawn.cs
using UnityEngine;

public class _ObjectSpawn : MonoBehaviour {
	public bool spawning = false;
	private GameObject newEnemy;
	private int canSpawn = 1;
	public int player = 1;
	public float minSpawnPerSecond = 0.3f;
	public float maxSpawnPerSecond = 0.6f;
	public GameObject[] Enemy;
	public _PlayerStats playerStats;

	public GameObject bgA;
	public GameObject bgB;
	public _bgControl bgScriptA;
	public _bgControl bgScriptB;

	void Start() {
		playerStats = gameObject.GetComponent<_PlayerStats>();
		bool logic;
		
		if (player == 1) {
			logic = false;
		} else {
			logic = true;
		}
		
		bgScriptA = bgA.GetComponent<_bgControl>();
		bgScriptA.dirRight = logic;
		bgScriptB = bgB.GetComponent<_bgControl>();
		bgScriptB.dirRight = logic;	
	}
	
	void FixedUpdate() {
		canSpawn -= 1;
		
		if (spawning && canSpawn <= 0) {
			int randObstacle = Random.Range(0, Enemy.Length);
			int randPos = Random.Range(1, 4);
			float newPos = 0;
			
			switch (randPos) {
				case 1:
					newPos = -3.5f;
					break;
				case 2:
					newPos = -0.5f;
					break;
				case 3:
					newPos = 1.5f;
					break;
			}
			
			newEnemy = Instantiate(Enemy[randObstacle], new Vector3(0, newPos, 0), Quaternion.identity) as GameObject;
			_ObjectControl oc = null;
			float speedMulti = 0;
			
			if (player == 1) {
				speedMulti = playerStats.P1Speed / 10;
				oc = newEnemy.GetComponent<_ObjectControl>();
				oc.dirRight = false;
			} else {
				speedMulti = playerStats.P2Speed / 10;
				oc = newEnemy.GetComponent<_ObjectControl>();
				oc.dirRight = true;
			}
			
			oc = newEnemy.GetComponent<_ObjectControl>();
			oc.speed *= speedMulti;
			canSpawn = (int) Random.Range((25 / (maxSpawnPerSecond * speedMulti)), (25 / (minSpawnPerSecond * speedMulti)));
		}
		
		float newSpeed = 0;
		
		if (player == 1) {
			newSpeed = playerStats.P1Speed;
		} else{
			newSpeed = playerStats.P2Speed;
		}
		
		bgScriptA.speed = newSpeed;
		bgScriptB.speed = newSpeed;
	}
}
